src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java

```diff
@@ -692,93 +690,124 @@ public void scan(HttpMessage msg, String param, String origParamValue) {
-					} catch (SocketException ex) {
-						if (log.isDebugEnabled()) log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() + 
-								" when accessing: " + msg4.getRequestHeader().getURI().toString());
-						return; //Something went wrong, no point continuing
-					}
-					countExpressionBasedRequests++;
-
-					String modifiedExpressionOutputUnstripped = msg4.getResponseBody().toString();
-					String modifiedExpressionOutputStripped = this.stripOff(modifiedExpressionOutputUnstripped, modifiedParamValue);
-
-					//set up two little arrays to ease the work of checking the unstripped output, and then the stripped output
-					String normalBodyOutput[] = {mResBodyNormalUnstripped, mResBodyNormalStripped};
-					String expressionBodyOutput[] = {modifiedExpressionOutputUnstripped, modifiedExpressionOutputStripped};
-					boolean strippedOutput[] = {false, true};
+					for (int booleancheck4VariantIndex = 0; booleancheck4VariantIndex<2 && !sqlInjectionFoundForUrl && countExpressionBasedRequests < doExpressionMaxRequests; booleancheck4VariantIndex++) {
+						try {
+							
+							String modifiedParamValue = "";
 
-					 for (int booleanStrippedUnstrippedIndex = 0; booleanStrippedUnstrippedIndex < 2 && !sqlInjectionFoundForUrl; booleanStrippedUnstrippedIndex++) {
-						//if the results of the modified request match the original query, we may be onto something. 
-						if (expressionBodyOutput[booleanStrippedUnstrippedIndex].compareTo(normalBodyOutput[booleanStrippedUnstrippedIndex]) == 0) {
-							if (this.debugEnabled) {
-								log.debug("Check 4, " + (strippedOutput[booleanStrippedUnstrippedIndex] ? "STRIPPED" : "UNSTRIPPED") + " html output for modified expression parameter [" + modifiedParamValue + "] matched (refreshed) original results for " + refreshedmessage.getRequestHeader().getURI());
+							if (check4Variant[booleancheck4VariantIndex] == "plus"){
+								//and set the parameter value to a string value like "3-2", if the original parameter value was "1"
+								int paramPlusTwo = paramAsInt + 2;
+								modifiedParamValue = String.valueOf(paramPlusTwo) + "-2";
 							}
-							//confirm that a different parameter value generates different output, to minimise false positives
-
-							//get a value 3 sizes bigger this time
-							int paramPlusFour = paramAsInt + 3;
-							String modifiedParamValueConfirm = String.valueOf(paramPlusFour) + "-2";
-
-							//and prepare a request to set the parameter value to a string value like "4-2", if the original parameter value was "1"
-							//Note that the two values are NOT equivalent, and the param value is different to the original
-							HttpMessage msg4Confirm = getNewMsg();
-							setParameter(msg4Confirm, param, modifiedParamValueConfirm);
+							else if (check4Variant[booleancheck4VariantIndex] == "mult"){
+								//and set the parameter value to a string value like "2/2", if the original parameter value was "1"
+								int paramPlusTwo = paramAsInt * 2;
```

Comment:
The variable name should change, it isn't a "PlusTwo" at this point. Perhaps `increasedParamValue` or `modifiedParamValue` or something more generic like that to fit both variants?